### ISSUE TITLE
HTML widget header styles don't match IPython.display.HTML

### ISSUE THREAD

**Issue by jeffyjefflabs** [jupyter-widgets/ipywidgets#2813]
I don't know if they're *supposed* to match, but header (`<h1>` etc.) tags are styled differently in HTML widgets vs the rest of Jupyter.  Size, weight, and spacing (or alignment?) are all different.  IPython.display.HTML, IPython.display.Markdown, and rendered markdown cells all match each other.

Sample code:

```
import ipywidgets as widgets
from IPython.display import display, HTML

grid = widgets.GridspecLayout(6, 2)
for i in range(6):
    html = f"<h{i + 1}>Header {i + 1}</h{i + 1}>"
    grid[i, 0] = widgets.Output()
    with grid[i, 0]:
        display(HTML(html))
    grid[i, 1] = widgets.HTML(html)
grid
```

Output (Chromium on Ubuntu):

![image](https://user-images.githubusercontent.com/18747664/76171813-07407280-6155-11ea-8f75-f217961ad4bc.png)

For what it's worth, changing the class on the HTML widget's div from `widget-html-content` to `jp-RenderedHTMLCommon` in the browser console makes them match.

Backstory: we were migrating a dashboard from the classic jupyter_dashboards extension to Voila and noticed the fonts didn't match, so we started looking for workarounds.  (We still have people using Classic, so we wanted the dashboard to look the same in both.)  Turns out between widgets.HTML vs IPython.display, and Classic vs Lab/Voila, there are *four* different sets of styles being applied.

**Comment by vidartf** [comment]
> Turns out between widgets.HTML vs IPython.display, and Classic vs Lab/Voila, there are four different sets of styles being applied

I agree that our HTML and that of the HTML outputs should ideally have the same base styling. Letting different front-ends style their rendered HTML seems like their prerogative. The question then becomes how to best make this work (should ipywidgets add classes/styling to work with all known frontends, or should all frontends ensure that all outputs receive the same styling, or something in between?). I might lean towards this being best resolved in the frontends (e.g. all headers in outputs are styled the same way, irregardless of whether it is rendered HTML or not), but I can see there being arguments against this.

**Comment by dsblank** [comment]
It's even worse that what you describe:

![Screenshot from 2021-08-19 17-32-30](https://user-images.githubusercontent.com/168568/130160933-14eb64d8-8c91-4a7b-b9fc-e17e3705e387.png)

But your hint above helps: surround all output with:

```python
"""<div class="jp-RenderedHTMLCommon">""" + html_text + """</div>"""
```

**Comment by nthiery** [comment]
Thank you @dsblank for the tip! I had been wondering for a while why the html in my widgets
where missing style.

**Comment by nthiery** [comment]
Here is a little notebook to test this further:

[Uploading test-HTML-ipywidgets-ipython.ipynb.zip…]()

![image](https://user-images.githubusercontent.com/545328/227877034-85827b2b-0131-4a76-8b8f-847494f621ba.png)

**Comment by nthiery** [comment]
Btw: a variant of @dsblank tip: run `html.add_class('jp-RenderedHTMLCommon')` on the html
widget.
